Assets/Tests/PlayModeTests/ConsentTests.cs

```diff
@@ -55,6 +58,88 @@ public void TestDefaultStateOfConsents()
             AssertConsentAll(expectedValue: true);
         }
 
+        /// <summary>
+        /// Case: if 'RequiresConsent' isn't set in the configuration during initialization.
+        /// Result: Consent request should not send.
+        /// </summary>
+        [Test]
+        public void TestConsentsRequest_RequiresConsent_IsFalse()
+        {
+            CountlyConfiguration configuration = new CountlyConfiguration {
+                AppKey = _appKey,
+                ServerUrl = _serverUrl,
+            };
+
+            Countly.Instance.Init(configuration);
+
+            Assert.IsNotNull(Countly.Instance.Consents);
+            Assert.AreEqual(1, Countly.Instance.CrashReports._requestCountlyHelper._requestRepo.Count);
+
+            CountlyRequestModel requestModel = Countly.Instance.Consents._requestCountlyHelper._requestRepo.Dequeue();
+            string myUri = requestModel.RequestUrl;
+            string consents = HttpUtility.ParseQueryString(myUri).Get("consent");
+            Assert.IsNull(consents);
+
+            Countly.Instance.Consents.GiveConsent(new Consents[] { Consents.Sessions });
+            Assert.AreEqual(0, Countly.Instance.CrashReports._requestCountlyHelper._requestRepo.Count);
+
+            Countly.Instance.Consents.RemoveConsent(new Consents[] { Consents.Sessions });
+            Assert.AreEqual(0, Countly.Instance.CrashReports._requestCountlyHelper._requestRepo.Count);
+
+        }
+
+        /// <summary>
+        /// It validates the initial consent request that generates after SDK initialization
+        /// </summary>
+        [Test]
+        public void TestConsentRequest()
+        {
+            CountlyConfiguration configuration = new CountlyConfiguration {
+                AppKey = _appKey,
+                ServerUrl = _serverUrl,
+                RequiresConsent = true,
+            };
+
+            configuration.GiveConsent(new Consents[] { Consents.Crashes, Consents.Events, Consents.Clicks, Consents.StarRating, Consents.Views, Consents.Users, Consents.Push, Consents.RemoteConfig, Consents.Location });
+            Countly.Instance.Init(configuration);
+
+            Assert.IsNotNull(Countly.Instance.Consents);
+            Assert.AreEqual(1, Countly.Instance.CrashReports._requestCountlyHelper._requestRepo.Count);
+
+            CountlyRequestModel requestModel = Countly.Instance.Consents._requestCountlyHelper._requestRepo.Dequeue();
+            string myUri = requestModel.RequestUrl;
+            string consents = HttpUtility.ParseQueryString(myUri).Get("consent");
+            JObject json = JObject.Parse(consents);
+            Assert.IsTrue(json.GetValue("push").ToObject<bool>());
```

Comment:
Add an assert that validates how many keys are set in that json. Just to be sure that no others are there.